01-node-tutorial/answers/writeWithPromisesAwait.js

```diff
@@ -0,0 +1,32 @@
+const { writeFile, readFile } = require('fs').promises;
+
+// Writing three lines
+
+const writer = async () => {
+    try {
+        for (let i = 0; i < 1000; i++) {
+            await writeFile('./temporary/temp.txt', 'Hey man how are you doing\n', { flag: 'a' })
+        }
+    } catch (error) {
+        console.log(error)
+    }
+}
+
+
+// Reading the file
+
+const reader = async () => {
+    try {
+        const result = await readFile('./temporary/temp.txt', 'utf8');
+        console.log(result)
+    } catch(err) {
+        console.log(err)
+    }
+}
+
+const readWrite = async () => {
```

Comment:
# Optional Improvement
[Error Handling]
So in this section you were told to put the awaits in a try catch and you did, but there a nuance here with error handling that's less clear.
Because the `reader` function and `writer` function both try catch all possible errors, its impossible for either to throw an exception. 
This means if your code calls writer or reader they may fail, but the calling code doesn't know that they failed. The calling code may need to do something different if they fail, for instance here you call writer then call reader on lines 28 and 29.
If writer fails, you probably don't want to subsequently call reader. But your readWrite function has no way of knowing if writer was successful or not.

The better way to do this would be to remove the try/catch blocks from both reader and writer and add a single try/catch in readWrite.
```js 
const readWrite = async () => {
    try {
      await writer();
      await reader();
    } catch(err) {
      console.log(err);
    }
}
```

With this code, if the call to `await writer();` fails, the call to `await reader();` never happens.